src/snyk/snykOss/editor/editorDecorator.ts

```diff
@@ -0,0 +1,128 @@
+import _ from 'lodash';
+import { IVSCodeLanguages } from '../../common/vscode/languages';
+import { IThemeColorAdapter } from '../../common/vscode/theme';
+import {
+  DecorationOptions,
+  TextEditorDecorationType,
+  ThemableDecorationInstanceRenderOptions,
+} from '../../common/vscode/types';
+import { IVSCodeWindow } from '../../common/vscode/window';
+import { messages } from '../messages/vulnerabilityCount';
+import { ImportedModule, ModuleVulnerabilityCount } from '../services/vulnerabilityCount/importedModule';
+
+type LineDecorations = DecorationOptions[]; // array index is a line number
+
+export class EditorDecorator {
+  private readonly decorationType: TextEditorDecorationType;
+  private readonly fileDecorationMap: Map<string, LineDecorations>;
+
+  private updateTimeout: NodeJS.Timer | undefined = undefined;
+
+  constructor(
+    private readonly window: IVSCodeWindow,
+    private readonly languages: IVSCodeLanguages,
+    private readonly themeColorAdapter: IThemeColorAdapter,
+  ) {
+    this.fileDecorationMap = new Map<string, LineDecorations>();
+    this.decorationType = this.window.createTextEditorDecorationType({
+      after: { margin: '0 0 0 1rem' },
+    });
+  }
+
+  get fileDecorations(): ReadonlyMap<string, LineDecorations> {
+    return this.fileDecorationMap;
+  }
+
+  setScanStartDecorations(filePath: string, modules: ImportedModule[]): void {
+    const lineDecorations: LineDecorations = [];
+
+    for (const module of modules) {
+      if (module.line == null) {
+        continue;
+      }
+
+      lineDecorations[module.line] = {
+        range: this.languages.createRange(module.line - 1, 1024, module.line - 1, 1024),
+        renderOptions: this.getRenderOptions(messages.fetchingVulnerabilities),
+      };
+    }
+
+    if (!lineDecorations.length) {
+      // return early when no decorations have been created
+      return;
+    }
+
+    this.fileDecorationMap.set(filePath, lineDecorations);
+    this.triggerUpdateDecorations(filePath);
+  }
+
+  setScanDoneDecorations(filePath: string, vulnerabilityCounts: ModuleVulnerabilityCount[]): void {
+    for (const count of vulnerabilityCounts) {
```

Comment:
this naming with `count` is a bit confusing for me. Maybe `fileVulnerabilities`?

Reply:
👍